python/ray/tests/test_actor_failures.py

```diff
@@ -1209,6 +1162,82 @@ def ping(self):
     assert " Worker unexpectedly exits" not in str(exc_info.value)
 
 
+def test_actor_restart_and_actor_received_task(shutdown_only):
+    # Create an actor with max_restarts=1 and max_task_retries=1.
+    # Submit a task to the actor and kill the actor after it receives
+    # the task. Then, the actor should restart in another core worker
+    # process, and the driver should resubmit the task to the new process.
+    # The task should be executed successfully.
+    @ray.remote(max_restarts=1, max_task_retries=1)
+    class RestartableActor:
+        def __init__(self):
+            self.counter = 0
+
+        def sleep_and_increment(self, sleep_time, signal_actor):
+            ray.get(signal_actor.send.remote())
+            time.sleep(sleep_time)
```

Comment:
This should be another signal actor's wait.

Reply:
follow up: https://github.com/ray-project/ray/pull/52121